File: patterns/javascript-patterns/pattern80.js
// Pattern 80: ...

//    *
//   *
//  *
// *
//  *
//   *
//    *

const n = 7;
let x = Math.floor(n / 2) + 1;

for (let i = 1; i <= n; i++) {
    let row = '';

    for (let j = 1; j <= x; j++) {
        if (j === x) {
            row += '* ';
        } else {
            row += ' ';
        }
    }

    console.log(row);

    if (i <= n / 2) {
        x--;
    } else {
        x++;
    }
}

File: patterns/python-patterns/pattern80.py
# Pattern 80: ...

#    *
#   *
#  *
# *
#  *
#   *
#    *

n = 7
x = n // 2 + 1

for i in range(1, n + 1):
    for j in range(1, x + 1):
        if j == x:
            print("* ", end="")
        else:
            print(" ", end="")
    print()

    if i <= n // 2:
        x -= 1
    else:
        x += 1

